src/Simulation/Native/src/external/fusion.hpp

```diff
@@ -58,21 +65,58 @@ class Fusion{
 		handle_controls(empty_matrix, empty_vec, {}); // remove all current control qubits (this is a GLOBAL factor)
 	}
 	
+	IndexSet get_target_set() const& {
+		return target_set_;
+	}
+
+	ItemVector get_items() const& {
+		return items_;
+	}
+
+	IndexSet get_ctrl_set() const& {
+		return ctrl_set_;
+	}
+
+	Complex get_global_factor() const& {
+		return global_factor_;
+	}
+
+	static void remap_qubits(std::set<Index>& qubits, const std::unordered_map<unsigned, unsigned>& mapFromOldLocToNewLoc) {
+		std::set<Index> tempSet;
+		for (unsigned elem : qubits) {
+			if (mapFromOldLocToNewLoc.find(elem) != mapFromOldLocToNewLoc.end()) {
+				tempSet.insert(mapFromOldLocToNewLoc.at(elem));
+			}
+		}
+		qubits.swap(tempSet);
+	}
+
+	void update_target_set(std::unordered_map<unsigned, unsigned> mapFromOldLocToNewLoc) {
```

Comment:
My bad, in the code snippet I suggested for the refactor, I forgot to change these signatures to `const ...&`. Could you pleaes do it to avoid the extra copy?